sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -68,6 +70,21 @@
         Utils.simpleObjectMapper.registerModule(new AfterburnerModule());
     }
 
+    public  static HttpHeaders clone(HttpHeaders inputHeaders) {
+        HttpHeaders httpHeaders = new HttpHeaders();
+        inputHeaders.forEach(entry -> httpHeaders.put(entry.getName(), entry.getValue()));
```

Comment:
potential minor perf impact implication.

previously we used to pass the the old headers to the constructor of the new HashMap to allow reserving the required memory upfront. this way the underlying header map may grow multiple times.